application_sandbox/shader_16bit_storage/main.cpp

```diff
@@ -0,0 +1,376 @@
+
+    VkClearValue clear;
+    vulkan::MemoryClear(&clear);
+
+    VkRenderPassBeginInfo pass_begin = {
+        VK_STRUCTURE_TYPE_RENDER_PASS_BEGIN_INFO,  // sType
+        nullptr,                                   // pNext
+        *render_pass_,                             // renderPass
+        *frame_data->framebuffer_,                 // framebuffer
+        {{0, 0},
+         {app()->swapchain().width(),
+          app()->swapchain().height()}},  // renderArea
+        1,                                // clearValueCount
+        &clear                            // clears
+    };
+
+    cmdBuffer->vkCmdBeginRenderPass(cmdBuffer, &pass_begin,
+                                    VK_SUBPASS_CONTENTS_INLINE);
+
+    cmdBuffer->vkCmdBindPipeline(cmdBuffer, VK_PIPELINE_BIND_POINT_GRAPHICS,
+                                 *cube_pipeline_);
+    cmdBuffer->vkCmdBindDescriptorSets(
+        cmdBuffer, VK_PIPELINE_BIND_POINT_GRAPHICS,
+        ::VkPipelineLayout(*pipeline_layout_), 0, 1,
+        &frame_data->cube_descriptor_set_->raw_set(), 0, nullptr);
+    cube_.Draw(&cmdBuffer);
+    cmdBuffer->vkCmdEndRenderPass(cmdBuffer);
+
+    (*frame_data->command_buffer_)
+        ->vkEndCommandBuffer(*frame_data->command_buffer_);
+  }
+
+  virtual void Update(float time_since_last_render) override {
+    model_data_->data().transform =
+        model_data_->data().transform *
+        Mat44::FromRotationMatrix(
+            Mat44::RotationX(3.14f * time_since_last_render) *
+            Mat44::RotationY(3.14f * time_since_last_render * 0.5f));
+    uint16_t v = uint16_t(std::abs((int32_t(frame_count) % 512) - 255));
+    colour_data_->data().r = v;
+    colour_data_->data().g = v;
+    colour_data_->data().b = v;
+  }
+  virtual void Render(vulkan::VkQueue* queue, size_t frame_index,
+                      CubeFrameData* frame_data) override {
+    // Update our uniform buffers.
+    camera_data_->UpdateBuffer(queue, frame_index);
+    model_data_->UpdateBuffer(queue, frame_index);
+    colour_data_->UpdateBuffer(queue, frame_index);
+
+    VkSubmitInfo init_submit_info{
+        VK_STRUCTURE_TYPE_SUBMIT_INFO,  // sType
+        nullptr,                        // pNext
+        0,                              // waitSemaphoreCount
+        nullptr,                        // pWaitSemaphores
+        nullptr,                        // pWaitDstStageMask,
+        1,                              // commandBufferCount
+        &(frame_data->command_buffer_->get_command_buffer()),
+        0,       // signalSemaphoreCount
+        nullptr  // pSignalSemaphores
+    };
+
+    app()->render_queue()->vkQueueSubmit(app()->render_queue(), 1,
+                                         &init_submit_info,
+                                         static_cast<VkFence>(VK_NULL_HANDLE));
+
+    frame_count++;
+  }
+
+ private:
+  struct CameraData {
+    Mat44 projection_matrix;
+  };
+
+  struct ModelData {
+    Mat44 transform;
+  };
+
+  struct ColourData {
```

Comment:
We use both color and colour in the sample. We should be consistent.

Reply:
Done, thanks!